podman/domain/config.py

```diff
@@ -64,15 +69,22 @@ def __init__(self, path: Optional[str] = None):
 
         if path is None:
             home = Path(xdg.BaseDirectory.xdg_config_home)
-            self.path = home / "containers" / "containers.conf"
+            self.path = home / "containers" / "podman-connections.json"
+            old_toml_file = home / "containers" / "containers.conf"
         else:
             self.path = Path(path)
 
         self.attrs = {}
         if self.path.exists():
-            with self.path.open(encoding='utf-8') as file:
+            with open(self.path, encoding='utf-8') as file:
+                self.attrs = json.load(file)
+
+        # Read the old toml file configuration
+        if old_toml_file.exists():
+            with old_toml_file.open(encoding='utf-8') as file:
                 buffer = file.read()
-            self.attrs = toml_loads(buffer)
+            loaded_toml = toml_loads(buffer)
+            self.attrs.update(loaded_toml)
```

Comment:
I think you should load the toml file, then update the dict with results of the json load.

Reply:
https://github.com/containers/podman-py/pull/374/files#diff-891006306a18924b730354f5400759161db17c6e727057a5fa71ef90d525a3b9R130 is where we create the services we return to the user. The `__init__` function just reads the two files and loads the keys and values into the `self.attrs` dict.